ttnn/cpp/ttnn/operations/eltwise/ternary/where_op.cpp

```diff
@@ -45,46 +45,46 @@ inline Tensor _where_op(
 }
 
 
-Tensor WhereOp::_where(
+Tensor WhereOperation::operator()(
     uint8_t queue_id,
     const Tensor& predicate,
     const Tensor& value_true,
     const Tensor& value_false,
     const std::optional<MemoryConfig>& output_mem_config,
     std::optional<Tensor> output_tensor) {
 
-    return _where_op(queue_id, predicate, value_true, value_false, output_mem_config, output_tensor);
+    return _where_op(queue_id, predicate, value_true, value_false, output_mem_config.value_or(predicate.memory_config()), output_tensor);
 }
 
-Tensor WhereOp::_where(
+Tensor WhereOperation::operator()(
     uint8_t queue_id,
     const Tensor& predicate,
     const float value_true,
     const Tensor& value_false,
     const std::optional<MemoryConfig>& output_mem_config,
     std::optional<Tensor> output_tensor) {
 
-    return _where_op(queue_id, predicate, value_true, value_false, output_mem_config, output_tensor);
+    return _where_op(queue_id, predicate, value_true, value_false, output_mem_config.value_or(predicate.memory_config()), output_tensor);
```

Comment:
- Rename _where_op to `where_impl` 
- Place it into an anonymous namespace in this file
- Remove inline keyword from it